### ISSUE TITLE
Ensure pre-commit hooks for client-side changes use package.json versions

### ISSUE THREAD

**Issue by lb-** [wagtail/wagtail#10755]
## Problem

Currently, any updates to Eslint packages, Stylelint packages and Prettier have to be manually kept in sync with the pre-commit hooks config.

Ideally we should not have to do this and the pre-commit hooks should use the same scripts / versions that are set up in package.json.

## Idea

> In theory you could make use of `repo: local` - https://pre-commit.com/#repository-local-hooks and use it something like: https://github.com/pre-commit/pre-commit/issues/945#issuecomment-527603460 ([`npm run lint:format`](https://github.com/wagtail/wagtail/blob/main/package.json#L140))

_Originally posted by @zerolab in https://github.com/wagtail/wagtail/pull/10742#discussion_r1283313860_

**Comment by thibaudcolas** [comment]
And I see our pre-commit hooks for stylelint are out-of-date again after #10719 🙈

**Comment by lb-** [comment]
OK I will close this. It's super tedious to keep these manually up to date but I will have a PR up to update this shortly.

**Comment by laymonage** [comment]
@lb- I thought Thibaud's comment above meant that this would be useful? I'm in favour of this so I don't mind us giving it a try after we get #11595 in.

**Comment by lb-** [comment]
Note that https://github.com/wagtail/wagtail/pull/11679 has been raised to further fix the manual sync of package version updates with the pre-commit config. 

Looks like there was a typo (on my part) but additional inconsistencies between the latest available packages and what the 'mirror' packages have in their repo for Prettier and others. 

There must be a better way to manage this, it's error prone and stops us from keeping our core code up to date due to the mirror packages also needing updating. 

Hopefully I'll get a chance to understand this stuff better, but anyone with better knowledge of pre-commit can chime in with some ideas.

**Comment by lb-** [comment]
These got out of sync again.

Fixed by @laymonage  https://github.com/wagtail/wagtail/pull/13223

I am going to try something out to see if we can get this check in our CI.

**Comment by lb-** [comment]
OK, I have an approach we can use, I will raise a separate PR, here's a quick snapshot of what the CI will look like if these get out of sync.

```
Checking package dependencies against `.pre-commit-config.yaml`.
file:///home/circleci/project/scripts/check-dependencies.mjs:58
  throw new Error(
        ^

Error: The following 'package.json' dependencies are not in sync with '.pre-commit-config.yaml':
prettier: 3.1.0 (package.json: ^3.6.2) [hook]
prettier: 3.3.3 (package.json: ^3.6.2)
eslint: 8.56.0 (package.json: ^8.57.0) [hook]
eslint: 8.56.0 (package.json: ^8.57.0)
@wagtail/eslint-config-wagtail: 0.4.0 (package.json: ^0.5.0)

    at file:///home/circleci/project/scripts/check-dependencies.mjs:58:9
    at ModuleJob.run (node:internal/modules/esm/module_job:268:25)
    at async onImport.tracePromise.__proto__ (node:internal/modules/esm/loader:543:26)
    at async asyncRunEntryPointWithESMLoader (node:internal/modules/run_main:116:5)

Node.js v22.11.0

Exited with code exit status 1
```

<img width="1429" height="686" alt="Image" src="https://github.com/user-attachments/assets/310f7b53-3e4c-4362-b8de-9d95ee130527" />

**Comment by lb-** [comment]
PR up with a CI driven approach to check these are in sync.

https://github.com/wagtail/wagtail/pull/13228

FYI @thibaudcolas  @laymonage

**Comment by gasman** [comment]
Fixed in #13228
